File: pythonlibrary/aibrd/extractors/acceptance_criteria.py
from ..llm.client import call_llm_json

_SYSTEM = """You are a QA analyst extracting acceptance criteria in Given/When/Then format from a BRD.
Return JSON: {"criteria": [{"given": "...", "when": "...", "then": "..."}]}"""


def extract_acceptance_criteria(brd_text: str) -> list[dict]:
    result = call_llm_json(brd_text, _SYSTEM)
    return result.get("criteria", [])

File: vscodebase/src/core/extractors/acceptance_criteria.ts
import * as vscode from 'vscode'
import { callLLMJson } from '../../llm/client'
import { AcceptanceCriteria } from '../models/outputs'

const SYSTEM = `You are a QA analyst extracting acceptance criteria from a BRD in Given/When/Then format.
For each business flow or feature, derive the key acceptance criteria.
Return JSON: {
  "criteria": [{ "given": "...", "when": "...", "then": "..." }]
}`

export async function extractAcceptanceCriteria(
  brdText: string,
  token?: vscode.CancellationToken
): Promise<Omit<AcceptanceCriteria, 'id' | 'relatedFlow' | 'relatedRules'>[]> {
  const result = await callLLMJson<{
    criteria: Omit<AcceptanceCriteria, 'id' | 'relatedFlow' | 'relatedRules'>[]
  }>(brdText, SYSTEM, token)
  return result.criteria
}
